src/medCore/gui/medSelectorWorkspace.cpp

```diff
@@ -37,13 +36,23 @@ medAbstractWorkspace(parent), d(new medSelectorWorkspacePrivate)
     selectorToolBox()->setTitle(name); // get workspace name
 }
 
+medSelectorWorkspace::medSelectorWorkspace(QWidget * parent, QString name, medSelectorToolBox* toolbox) : medAbstractWorkspace(parent), d(new medSelectorWorkspacePrivate)
+{
+    d->selectorToolBox = toolbox;
+
+    connect(toolbox,SIGNAL(success()),this,SLOT(onProcessSuccess()));
+
+    this->addToolBox(toolbox);
+    toolbox->setTitle(name); // get workspace name
```

Comment:
The duplicate code between constructors can be factorised into an initialize function or similar, or you could change the signature of this constructor to set the default value of toolbox to NULL, in which case it will be initialised to a standard selector toolbox.